relayer/aztec/relayer.go

```diff
@@ -0,0 +1,380 @@
+	const abiJSON = `[{
+        "inputs": [{"internalType": "bytes", "name": "encodedVM", "type": "bytes"}],
+        "name": "parseAndVerifyVM",
+        "outputs": [
+            {"internalType": "tuple", "name": "", "type": "tuple"},
+            {"internalType": "bool", "name": "", "type": "bool"},
+            {"internalType": "string", "name": "", "type": "string"}
+        ],
+        "stateMutability": "view",
+        "type": "function"
+    }]`
+
+	parsedABI, err := abi.JSON(strings.NewReader(abiJSON))
+	if err != nil {
+		return false, fmt.Errorf("ABI parse error: %w", err)
+	}
+
+	data, err := parsedABI.Pack("parseAndVerifyVM", vaaBytes)
+	if err != nil {
+		return false, fmt.Errorf("ABI pack error: %w", err)
+	}
+
+	targetAddr := common.HexToAddress(targetContract)
+	msg := ethereum.CallMsg{
+		To:   &targetAddr,
+		Data: data,
+	}
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	result, err := c.client.CallContract(ctxWithTimeout, msg, nil)
+	if err != nil {
+		return false, fmt.Errorf("contract call failed: %w", err)
+	}
+
+	// Extract the "valid" boolean from the result
+	if len(result) >= 64 {
+		return result[63] == 1, nil // Check the last byte of the second 32-byte word
+	}
+
+	return false, fmt.Errorf("unexpected result format")
+}
+
+// Relayer coordinates processing VAAs from the spy service
+type Relayer struct {
+	spyClient    *SpyClient
+	evmClient    *EVMClient
+	config       Config
+	vaaProcessor func(*Relayer, *VAAData) error
+}
+
+// NewRelayer creates a new relayer instance
+func NewRelayer(config Config) (*Relayer, error) {
+	log.Println("[Relayer] Initializing...")
+
+	// Connect to the spy service
+	spyClient, err := NewSpyClient(config.SpyRPCHost)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create spy client: %w", err)
+	}
+
+	// Connect to the EVM chain
+	evmClient, err := NewEVMClient(config.DestRPCURL, config.PrivateKey)
+	if err != nil {
+		spyClient.Close()
+		return nil, fmt.Errorf("failed to create EVM client: %w", err)
+	}
+
+	relayer := &Relayer{
+		spyClient: spyClient,
+		evmClient: evmClient,
+		config:    config,
+	}
+
+	// Set default VAA processor
+	relayer.vaaProcessor = defaultVAAProcessor
+
+	return relayer, nil
+}
+
+// Close cleans up resources used by the relayer
+func (r *Relayer) Close() {
+	if r.spyClient != nil {
+		r.spyClient.Close()
+	}
+}
+
+// Start begins listening for VAAs and processing them
+func (r *Relayer) Start(ctx context.Context) error {
+	log.Printf("[Relayer] Starting with address: %s", r.evmClient.GetAddress().Hex())
+	log.Printf("[Relayer] Filtering for VAAs from chain %d to chain %d", r.config.SourceChainID, r.config.DestChainID)
+	log.Printf("[Relayer] Monitoring emitter address: %s", r.config.EmitterAddress)
+
+	// Subscribe to VAAs
+	stream, err := r.spyClient.SubscribeSignedVAA(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to subscribe to VAA stream: %w", err)
```

Comment:
nit: it is generally preferred to form error messages like `"action: %v"`. Using `%w` makes the wrapped error become part of the API, which we should use sparingly. For example, here we can use `"subscribe to vaa: %v"`. This way, if many error message are combined as we climb back up the call stack, we get something like 
```
Relayer failure: foo: bar: subscribe to vaa
```
Instead of repeating `failed to` many times:
```
Relayer failure: failed to foo: failed to bar: failed to subscribe to vaa
```